fast64_internal/sm64/sm64_texscroll.py

```diff
@@ -122,24 +138,28 @@ def writeTexScrollBase(baseDir):
 	scrollData = texscrollCFile.read()
 	texscrollCFile.close()
 
+	texScrollIncludeDef = '#include "texscroll.h"'
+	macroIndex = scrollData.index(texScrollIncludeDef)
+
+	if '#include "tile_scroll.h"' not in scrollData:
+		scrollData = scrollData[:macroIndex] + '#include "tile_scroll.h"\n' + scrollData[macroIndex:]
+		macroIndex = scrollData.index(texScrollIncludeDef)
+
 	scrollConditionDefine = '#ifdef TARGET_N64\n' +\
 		'#define SCROLL_CONDITION(condition) condition\n' +\
 		'#else\n' +\
 		'#define SCROLL_CONDITION(condition) 1\n' +\
 		'#endif\n'
 	if '#define SCROLL_CONDITION' not in scrollData:
-		texScrollIncludeDef = '#include "texscroll.h"'
-		macroIndex = scrollData.index(texScrollIncludeDef)
 		if macroIndex != -1:
 			macroIndex += len(texScrollIncludeDef)
 			scrollData = scrollData[:macroIndex] + '\n\n' + scrollConditionDefine +\
 				scrollData[macroIndex:]
 		else:
 			raise PluginError("Cannot find '#include \"texscroll.h\" in src/game/texscroll.c")
 
-		texscrollCFile = open(texscrollCPath, 'w', newline='\n')
+	with open(texscrollCPath, 'w', newline='\n') as texscrollCFile:
 		texscrollCFile.write(scrollData)
-		texscrollCFile.close()
 	
```

Comment:
did you intend for this to not be under the if block anymore?

Reply:
ahhh good catch! yup that should be in there

Reply:
updated!